File: hot-framework/src/test/resources/mongo.js
db.getCollection("posts").drop()

var post = {"author": "Mike",
		"text": "My first blog post!",
		"tags": ["mongodb", "python", "pymongo"]}

var post2 = {"author": "Julie",
		"text": "My first blog post!",
		"tags": ["mongodb", "python", "pymongo"]}

var post3 = {"author": "Damien",
		"text": "My first blog post!",
		"tags": ["mongodb", "python", "pymongo"],
		"doc": {
			"title":"doc1",
			"writer":"julie"
			}
		}

db.getCollection("posts").insert(post)
db.getCollection("posts").insert(post2)
db.getCollection("posts").insert(post3)

_.each (db.getCollection("posts").find().toArray(),function (obj) {
	out.println(obj["_id"]+" title: "+ obj["text"]+ ", tag: " +obj["tags"][0]);
	if (obj.doc != null) {
		out.println("val:" +obj.doc.title+"");
	}
})

out.println (db.getCollection("posts").findOne().author);
out.println (db.getCollection("posts").findOne({"author": "Mike"})._id);

out.println (db.getCollection("posts").find({"author": "Eliot"}).count());
out.println (db.getCollection("posts").find({"author": "Mike"}).count());

var criterion = {"_id": db.getCollection("posts").findOne({"author": "Mike"})._id}
out.println (db.getCollection("posts").find(criterion).count());
db.getCollection("posts").update({"author": "Toto","text":"hello world"},criterion)
db.getCollection("posts").findOne({"author": "Mike"})

File: hot-framework/src/test/resources/mongo.py
from com.mongodb import *
from be.solidx.hot.data.mongo import *

class DB:
	def __init__ (self,db):
		self.db = db
		
	def getCollection(self, name):
		return BasicCollection(self.db,name,DBObjectPyDictionaryTransformer())

	def __getattr__(self, name):
   		return self.getCollection(name)

mongo = Mongo()
db = DB(mongo.getDB("TestMongoTemplat"))


post = {"author": "Mike",
		"text": "My first blog post!",
		"tags": ["mongodb", "python", "pymongo"]}

post2 = {"author": "Julie",
		"text": "My first blog post!",
		"tags": ["mongodb", "python", "pymongo"]}
		
post3 = {"author": "Julie",
		"text": "My first blog post!",
		"tags": ["mongodb", "python", "pymongo"],
		"doc": {
			"title":"doc1",
			"writer":"julie"
			}
		}

db.posts.drop()		

db.posts.insert(post)
db.posts.insert(post2)
db.posts.insert(post3)

print db.posts.findOne()
print db.posts.findOne({"author": "Mike"})
db.posts.findOne({"author": "Eliot"})

for post in db.posts.find():
	print post

for post in db.posts.find({"author": "Mike"}):
	print post

for post in db.posts.find().sort({"author":1}):
	print post

print db.posts.find({"author": "Mike"}).count()
db.posts.update({"author":"Toto"},{"author":"Mike"})
print db.posts.find().count()
for post in db.posts.find().sort({"author":1}):
	print post